### DISCUSSION TITLE
自定义 layout 获取当前的路由信息

### DISCUSSION THREAD

**Discussion by zFitness** [umijs/umi#11030] [General]
如何在用户管理2的页面获取到它下面的 routes呢
```
          {
            name: '用户管理2',
            path: '/platform/user-list2',
            component: '@/components/tab-layout',
            routes: [
              {
                name: '用户管理2-1',
                hideInMenu: true,
                path: '/platform/user-list2/user-list2-1',
                component: 'platform/employee',
              },
              {
                name: '用户管理2-2',
                hideInMenu: true,
                path: '/platform/user-list2/user-list2-2',
                component: 'platform/employee',
              },
            ],
          },
```

**Comment by fz6m** [comment]
[useAppData](https://umijs.org/docs/api/api#useappdata) 或 [useSelectedRoutes](https://umijs.org/docs/api/api#useselectedroutes) 里面找找数据。

**Reply by zFitness** [reply]
现在已经获取到了：
```
  const route = useRouteData();
  const selectRoutes = useSelectedRoutes();
  const layoutPath = route.route.path;
  const routeProps = useRouteProps();

  const layoutChildren = useMemo(() => {
    if (!layoutPath) return [];
    return (
      selectRoutes.find((item) => item.pathname === layoutPath)?.route
        ?.children || []
    );
  }, [layoutPath]);
```

请问这里 useRouteData 和 useSelectedRoutes的区别是啥

**Reply by fz6m** [reply]
`useRouteData` 是个早期的 api ，为了匹配出当前路由的简单信息，但有两个缺点，一是嵌套路由没法知道自己在哪条路径上，二是信息太少，所以后面 `useSelectedRoutes` 是更好的方案，他的最后一个元素就是当前路由，前面的就是嵌套的父路由，信息更全。

如果需要完整的路由表信息，用 `useAppData` 。

**Reply by zFitness** [reply]
大佬，为什么这里我访问 `/platform/user-list2` 和 `/platform/user-list2/user-list2-1`时，useRouteData都是返回的 `/platform/user-list2` 的路由，虽说我觉得这里还蛮实用的，但还是很迷惑。

第二个问题是访问`/platform/user-list2`页面时，useSelectedRoutes 返回的数组最后一个元素是 `/platform/user-list2` ; 访问 `/platform/user-list2/user-list2-1` 返回的数组最后一个元素是 `/platform/user-list2/user-list2-1`。这样我只使用这个 API 我就无法判断谁是父路由了。

**Reply by fz6m** [reply]
需要给一个最小复现，光描述不清楚上下文。
